main/src/ca/uwaterloo/flix/language/phase/Simplifier.scala

```diff
@@ -222,7 +237,7 @@ object Simplifier {
       throw InternalCompilerException(s"Unexpected expression: $exp0.", loc)
   }
 
-  private def visitType(tpe: Type)(implicit flix: Flix): MonoType = {
+  private def visitType(tpe: Type)(implicit flix: Flix, root: MonoAst.Root): MonoType = {
```

Comment:
We should pass the root as an implicit parameter (everywhere you use it here)

Reply:
> We should pass the root as an implicit parameter (everywhere you use it here)

Is that not already what I do?

Reply:
Doh, my bad. I was confused by the order. We always go from most specific to least specific, so it should be root: .., flix: Flix.

This is just by convention, but useful when you have many implicit parameters.